Packages/com.unity.inputsystem/InputSystem/Controls/InputControlExtensions.cs

```diff
@@ -808,5 +809,337 @@ internal static string BuildPath(this InputControl control, string deviceLayout,
 
             return builder.ToString();
         }
+
+        // Undocumented APIs. Meant to be used only by auto-generated, precompiled layouts.
+        // These APIs exist solely to keep access to the various properties/fields internal
+        // and only allow their contents to be modified in a controlled manner.
+        #region Undocumented
```

Comment:
This one's a compromise. The generated classes need a way to mutate data that is otherwise considered immutable. So far, this was not a problem as `InputDeviceBuilder` was the only place that required access. But the code generated for precompiled layouts may not be part of the input system and thus may not have `internal`-level access (the test I added in `CoreTests_Editor` sets up such a scenario).

However, I also did not want to simply expose all the properties/fields for arbitrary mutation. So I chose to have a "builder" kind of API that has time-limited write access which is turned off as soon the setup of a device is completed.

I added `AggressiveInlining` annotations to attempt to get rid of the compiler just generating tons and tons of function call overhead for what really is simple field writes.